### PR TITLE
1339 responsive readme

### PR DESCRIPTION
**Pull request by exbotanical**
This PR encompasses two primary changes: the media query hook, and the actual README changes.

1) I've written a custom hook,`useMedia`, which empowers the utility of media queries in component logic, effectively making conditional rendering for mobile accessible anywhere in the codebase without first having to instantiate state + lifecycle methods to determine innerWidth. 

It's built such that it can easily be modified to accept an array of breakpoints so as to handle multiple media queries in a single component (if ever needed). Other props could probably be passed in, too. 

2) I've also included a 'collapsible', styled wrapper component. The goal here was to have a small preview of the README text display on load, then allow toggling the remainder with the collapsible. 

Edit: Note that I've removed 'edit' buttons on mobile (only) so as to conserve visual real estate on dataset pages, per Eric's request.

IMPT: 

- JSDOM does not yet cover certain methods, window.matchMedia among them. [Their official docs suggest mocking these methods](https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom) in the pertinent test file, so I've done that here. 

- CircleCI ran into the same failure that has been occurring across many PRs of late - the dataset-uploaded snapshot contains the incorrect diction (i.e. 'almost' as opposed to 'about'). I've changed this, but want to note [the same change has been implemented in my other PR](https://github.com/OpenNeuroOrg/openneuro/pull/1373/commits/52b37a74ae632ecbd8ca14a6f923430cf1cee20f).

### CODE REVIEW

File: packages/openneuro-app/src/scripts/datalad/dataset/dataset-description.jsx [hunk 1]
```diff
@@ -8,15 +8,21 @@ const arrayToMarkdown = arr => {
   return arr ? arr.map(element => ` * ${element}\n`).join('') : ''
 }
 
-const DatasetDescription = ({ datasetId, description, editMode = false }) => (
+const DatasetDescription = ({
+  datasetId,
+  description,
+  isMobile,
```
Comment (david-nishi) [inline packages/openneuro-app/src/scripts/datalad/dataset/dataset-description.jsx:14]: This is missing from propTypes; breaking linter.

**Review by david-nishi — CHANGES_REQUESTED** [review]

File: packages/openneuro-app/src/scripts/mobile/collapsible-wrapper.jsx [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  const [isOpen, setOpen] = useState(false)
+  const toggleItem = () => {
+    setOpen(prevState => !prevState)
+  }
+  return (
+    <>
+      <Body>
+        <Collapse isOpen={isOpen}>{children}</Collapse>
+      </Body>
+      {!isOpen && (
```
[the remark this answers is not in the document]
Comment (exbotanical) [reply packages/openneuro-app/src/scripts/mobile/collapsible-wrapper.jsx]: > I think we should move this ternary internal to title square bracket accessors.

Excellent point. Doneskies.

File: packages/openneuro-app/src/scripts/mobile/collapsible-wrapper.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+import PropTypes from 'prop-types'
+import styled from '@emotion/styled'
+
+// wrapper for collapsible content on mobile
+
+const Content = styled.div`
+  height: 7em;
+  overflow: hidden;
+  text-align: left;
+  font-size: 14px;
```
[the remark this answers is not in the document]
Comment (exbotanical) [reply packages/openneuro-app/src/scripts/mobile/collapsible-wrapper.jsx]: Okay. Accidentally left that 14px in there when copying over link styles. I've just removed it; the show more/less text is now the same size as other text using that same font (seems to be the proverbial 'do something' typefont across OpenNeuro). 

Side note: I realized that the collapsible needs to be omitted when README content is null, lest we get an arbitrary gap and confusing 'show more/less' button with no text. To that end, I added a content (props) check to the condition in `dataset-readme`. Just a heads up.

**Comment by exbotanical** [comment]
Note: I realized that the collapsible was being rendered even when there was no README content. The problem this caused was the collapsible still afforded whitespace and thus would collapse to display "show less" and - while ostensibly fully expanded - display "show more". I added [a condition that checks for content in the README before rendering the collapsible wrapper. ](https://github.com/OpenNeuroOrg/openneuro/pull/1363/commits/bad7ca53babc12071db9369313c180bc819209db)

**Review by david-nishi — APPROVED** [review]
Looks good!

**Pull request merged**